src/Core/Content/Cms/Events/CmsPageLoadedEvent.php

```diff
@@ -7,30 +7,30 @@
 use Shopware\Core\Framework\DataAbstractionLayer\EntityCollection;
 use Shopware\Core\Framework\Event\NestedEvent;
 use Shopware\Core\Framework\Event\ShopwareSalesChannelEvent;
+use Shopware\Core\Framework\Feature;
 use Shopware\Core\Framework\Log\Package;
 use Shopware\Core\System\SalesChannel\SalesChannelContext;
 use Symfony\Component\HttpFoundation\Request;
 
 #[Package('discovery')]
 class CmsPageLoadedEvent extends NestedEvent implements ShopwareSalesChannelEvent
 {
-    protected Request $request;
-
-    protected CmsPageCollection $result;
-
-    protected SalesChannelContext $salesChannelContext;
-
     /**
+     * @deprecated tag:v6.8.0 - $result type will be changed from EntityCollection to CmsPageCollection
+     *
      * @param CmsPageCollection $result
      */
     public function __construct(
-        Request $request,
-        EntityCollection $result,
-        SalesChannelContext $salesChannelContext
+        protected Request $request,
+        protected EntityCollection $result,
+        protected SalesChannelContext $salesChannelContext,
     ) {
-        $this->request = $request;
-        $this->result = $result;
-        $this->salesChannelContext = $salesChannelContext;
+        if (!$this->result instanceof CmsPageCollection) {
```

Comment:
is that necessary? I would leave the normal property. passing the wrong type will now already fail with a type error :grin: not need to widen that now

Reply:
Mh yeah, I am just not sure, that it will then be made a promoted property.

However I can also change it, but I think it does not matter so much, and resolving the deprecations will be easier?